src/js/with-react.jsx

```diff
@@ -40,9 +39,9 @@ const isEventFilter = (key) => {
  * );
  */
 const withReact = (WebComponent, { pure = true, passive = false } = {}) => {
-  const name = functionName(WebComponent);
-  const displayName = `${name}React`;
-  const WCTagName = dasherize(name);
+  const { tagName } = (new WebComponent());
```

Comment:
wow ok i see

Reply:
but breaks with **Uncaught TypeError: Illegal constructor** 😞 
<img width="545" alt="screen shot 2018-05-04 at 15 01 47" src="https://user-images.githubusercontent.com/914443/39629310-059a3e46-4fac-11e8-9f18-8a26d0c98055.png">

Reply:
:(

Reply:
Okay, I quickly test two cases:
1. Instantiating `AXAButton` before it's definition by ` window.customElements.define('axa-button', AXAButton);` -> **Uncaught TypeError: Illegal constructor**
2.  Instantiating `AXAButton` after it's definition -> works